### PR TITLE
feat: add advance problem select

### PR DESCRIPTION
**Pull request by KristinAoki**
JIRA Ticket: [TNL-10216](https://2u-internal.atlassian.net/browse/TNL-10216)

This PR adds the functionality for selecting advance problem types.

### CODE REVIEW

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/index.jsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ SelectTypeWrapper.defaultProps = {
   onClose: null,
 };
 SelectTypeWrapper.propTypes = {
-  selected: PropTypes.func.isRequired,
+  selected: PropTypes.string.isRequired,
```
Comment (rayzhou-bit) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/index.jsx:46]: good catch

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/content/ProblemTypeSelect.jsx [hunk 2]
```diff
@@ -1,35 +1,43 @@
...
+import { ProblemTypes, ProblemTypeKeys, AdvanceProblemKeys } from '../../../../../data/constants/problem';
+import { FormattedMessage, injectIntl } from '@edx/frontend-platform/i18n';
+import messages from './messages';
 
 export const ProblemTypeSelect = ({
   selected,
   setSelected,
 }) => {
   const handleChange = e => setSelected(e.target.value);
+  const handleClick = () => setSelected('blankadvanced');
```
Comment (rayzhou-bit) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/content/ProblemTypeSelect.jsx:13]: use `AdvanceProblemKeys.BLANK`

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js [hunk 3]
```diff
@@ -15,7 +16,11 @@ export const selectHooks = () => {
   };
 };
 
-export const onSelect = ( setProblemType, selected ) => () => {
+export const onSelect = ( setProblemType, selected, updateField ) => () => {
+  if (Object.values(AdvanceProblemKeys).includes(selected)) {
+    console.log(AdvanceProblems[selected]);
+    updateField({rawOLX: AdvanceProblems[selected].template})
```
Comment (rayzhou-bit) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js:22]: We should just create a new reducer called `setOlxTemplate` or something. Much more readable and direct.

**Review by rayzhou-bit — CHANGES_REQUESTED** [review]
This looks great! Just have a couple thoughts and changes for you to look at.

- `services/cms/mockApi.js` should have a `fetchAdvanceSettings` function so this works in gallery view.

- Currently, `useArrowNav` which lets you move up and down selections by pressing the Up and Down arrow buttons will trigger an error. This can be used for `AdvancedTypeSelect` as well by accounting for `AdvanceProblems` in `useArrowNav` and adding `next`, `prev` properties on the `AdvanceProblems`. I could code this in if you like. I should have made it more error-proof.

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js (hunk 3 above)
Comment (KristinAoki) [reply src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js:22]: I don't think that creating another reducer makes the call more readable and direct. This is the only place the reducer would be used so it seems unnecessary to create a new reducer function for just one call. Unlike `setProblemType` which is probably used in multiple location.

Comment (rayzhou-bit) [reply src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js:21]: Ok, that makes sense to me. I was thinking about it from the point of view of reading an onSelect call and getting an understanding of what it's doing based on the inputs.

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js [hunk 4]
```diff
@@ -28,17 +32,25 @@ export const useArrowNav = (selected, setSelected) => {
...
+    const advanceTypeValues = Object.values(AdvanceProblemKeys);
     switch (e.key) {
       case 'ArrowUp':
-        if (ProblemTypes[selected].prev) {
+        if (problemTypeValues.includes(selected) && ProblemTypes[selected].prev) {
           setSelected(ProblemTypes[selected].prev);
           document.getElementById(ProblemTypes[selected].prev).focus();
+        } else if (advanceTypeValues.includes(selected) && AdvanceProblems[selected].prev) {
+          setSelected(AdvanceProblems[selected].prev);
+          document.getElementById(AdvanceProblems[selected].prev).focus();
```
Comment (rayzhou-bit) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js:44]: I just realized that Form.Radio has the arrow functionality coded in... I'm so sorry I did not realize that earlier. You could remove the `else if` with advanced types and arrow navigation will still work.

The type checking looks great.

File: src/editors/data/constants/problem.js [hunk 5]
```diff
@@ -27,6 +27,7 @@ export const ProblemTypes = StrictDict({
     preview: singleSelect,
     description: 'Specify one correct answer from a list of possible options',
     helpLink: 'https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/exercises_tools/multiple_choice.html',
+    prev: ProblemTypeKeys.TEXTINPUT,
```
Comment (rayzhou-bit) [inline src/editors/data/constants/problem.js:30]: Good catch. That will match how `Form.Radio` works.

**Review by rayzhou-bit — APPROVED** [review]

**Pull request merged**